### PR TITLE
refactor(generator): improve robustness of BuildNamespaces

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5169)

### CODE REVIEW

**Review by devbww — APPROVED** [review]

File: generator/internal/codegen_utils.cc [hunk 1]
```diff
@@ -75,24 +75,16 @@ std::string ProtoNameToCppName(absl::string_view proto_name) {
...
-                  "product_path must be present in vars.");
+std::vector<std::string> BuildNamespaces(std::string const& product_path,
+                                         NamespaceType ns_type) {
+  std::string name;
+  std::vector<std::string> v =
+      absl::StrSplit(product_path, '/', absl::SkipEmpty());
+  if (v.size() > 2) {
+    name = absl::StrJoin(v.begin() + 2, v.end(), "_");
+  } else {
+    name = absl::StrJoin(v.begin(), v.end(), "_");
```
Comment (devbww) [inline generator/internal/codegen_utils.cc:86]: Ergh.  It looks like I forgot to save the well-reasoned "nit" comment I wrote here, and now it has disappeared.

The short version is that I suggested ...
```
  std::string name = absl::StrJoin(v.begin() + (v.size() > 2 ? 2 : 0), v.end(), '_');
```
and
```
  absl::StrAppend(&name, "_internal");
```

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5169)* status: 0 of 5 files reviewed, 1 unresolved discussion (waiting on @devbww)

___
*[generator/internal/codegen_utils.cc, line 86 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5169#-MIQqvRdApUlJ19ktdQY-r1-86:-MIVAhFw0Whk8VP9ClTN:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/c95dba80f0097b0f0daea8ae82b458e2613bd983/generator/internal/codegen_utils.cc#L86)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Ergh.  It looks like I forgot to save the well-reasoned "nit" comment I wrote here, and now it has disappeared.

The short version is that I suggested ...
```
  std::string name = absl::StrJoin(v.begin() + (v.size() > 2 ? 2 : 0), v.end(), '_');
```
and
```
  absl::StrAppend(&name, "_internal");
```
</blockquote></details>

Done.
